src/plots/linePlot.ts

```diff
@@ -349,10 +374,65 @@ export module Plots {
 
     protected _getDataToDraw() {
       let dataToDraw = new Utils.Map<Dataset, any[]> ();
-      this.datasets().forEach((dataset) => dataToDraw.set(dataset, [dataset.data()]));
+
+      this.datasets().forEach((dataset) => {
+        let data = dataset.data();
+
+        if (!this._croppedRendering) {
+          dataToDraw.set(dataset, [data]);
+          return;
+        }
+
+        let filteredDataIndices = data.map((d, i) => i);
+
+        if (this._croppedRendering) {
+          filteredDataIndices = this._filterCroppedRendering(dataset, filteredDataIndices);
+        }
+
+        dataToDraw.set(dataset, [filteredDataIndices.map((d, i) => data[d])]);
+      });
+
       return dataToDraw;
     }
 
+    private _filterCroppedRendering(dataset: Dataset, indices: number[]) {
+      let xScale = this.x().scale;
+      let yScale = this.y().scale;
+      let xAccessor = this.x().accessor;
+      let yAccessor = this.y().accessor;
+      let xDomain = xScale.domain();
+      let yDomain = yScale.domain();
+
+      let data = dataset.data();
+      let filteredDataIndices: number[] = [];
+
+      for (let i = 0; i < indices.length; i++) {
+        let currXPoint = xAccessor(data[indices[i]], indices[i], dataset);
+        let currYPoint = yAccessor(data[indices[i]], indices[i], dataset);
+        let shouldShow = xDomain[0] <= currXPoint && currXPoint <= xDomain[1] &&
+          yDomain[0] <= currYPoint && currYPoint <= yDomain[1];
+
+        if (!shouldShow && indices[i - 1] != null && data[indices[i - 1]] != null) {
+          let prevXPoint = xAccessor(data[indices[i - 1]], indices[i - 1], dataset);
+          let prevYPoint = yAccessor(data[indices[i - 1]], indices[i - 1], dataset);
+          shouldShow = shouldShow || xDomain[0] <= prevXPoint && prevXPoint <= xDomain[1] &&
+            yDomain[0] <= prevYPoint && prevYPoint <= yDomain[1];
+        }
+
+        if (!shouldShow && indices[i + 1] != null && data[indices[i + 1]] != null) {
+          let nextXPoint = xAccessor(data[indices[i + 1]], indices[i + 1], dataset);
+          let nextYPoint = yAccessor(data[indices[i + 1]], indices[i + 1], dataset);
+          shouldShow = shouldShow || xDomain[0] <= nextXPoint && nextXPoint <= xDomain[1] &&
+            yDomain[0] <= nextYPoint && nextYPoint <= yDomain[1];
+        }
+
+        if (shouldShow) {
+          filteredDataIndices.push(indices[i]);
+        }
+      }
+      return filteredDataIndices;
```

Comment:
Instead of this code flow, why not check through the `width()` / `height()` pieces from the component?